cloud/services/loadbalancers.go

```diff
@@ -0,0 +1,115 @@
+package services
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"slices"
+	"strconv"
+
+	"github.com/go-logr/logr"
+	"github.com/linode/cluster-api-provider-linode/cloud/scope"
+	"github.com/linode/cluster-api-provider-linode/util"
+	"github.com/linode/linodego"
+)
+
+var (
+	defaultLBPort = 6443
+)
+
+// CreateNodeBalancer creates a new NodeBalancer if one doesn't exist
+func CreateNodeBalancer(ctx context.Context, clusterScope *scope.ClusterScope, logger logr.Logger) (*linodego.NodeBalancer, error) {
+	var linodeNBs []linodego.NodeBalancer
+	var linodeNB *linodego.NodeBalancer
+	NBLabel := fmt.Sprintf("%s-api-server", clusterScope.LinodeCluster.Name)
+	clusterUID := string(clusterScope.LinodeCluster.UID)
+	tags := []string{string(clusterScope.LinodeCluster.UID)}
+	filter := map[string]string{
+		"label": NBLabel,
+	}
+
+	rawFilter, err := json.Marshal(filter)
+	if err != nil {
+		return nil, err
+	}
+	if linodeNBs, err = clusterScope.LinodeClient.ListNodeBalancers(ctx, linodego.NewListOptions(1, string(rawFilter))); err != nil {
+		logger.Info("Failed to list NodeBalancers", "error", err.Error())
+
+		return nil, err
+	}
+	if len(linodeNBs) == 1 {
+		logger.Info(fmt.Sprintf("NodeBalancer %s already exists", *linodeNBs[0].Label))
+		if !slices.Contains(linodeNBs[0].Tags, clusterUID) {
+			err = errors.New("NodeBalancer conflict")
+			logger.Error(err, fmt.Sprintf("NodeBalancer %s is not associated with cluster UID %s. Owner cluster is %s", *linodeNBs[0].Label, clusterUID, linodeNBs[0].Tags[0]))
+
+			return nil, err
+		}
+
+		return &linodeNBs[0], nil
+	}
+
+	logger.Info(fmt.Sprintf("Creating NodeBalancer %s-api-server", clusterScope.LinodeCluster.Name))
+	createConfig := linodego.NodeBalancerCreateOptions{
+		Label:              util.Pointer(fmt.Sprintf("%s-api-server", clusterScope.LinodeCluster.Name)),
+		Region:             clusterScope.LinodeCluster.Spec.Region,
+		ClientConnThrottle: nil,
```

Comment:
I think we should be able to leave this off if we're not setting it or do `util.Pointer(0)` to be explicit